impl/src/main/java/org/jboss/weld/bean/proxy/ProxyFactory.java

```diff
@@ -174,28 +167,15 @@ public ProxyFactory(String contextId, Class<?> proxiedBeanType, Set<? extends Ty
         Class<?> superClass = typeInfo.getSuperClass();
         superClass = superClass == null ? Object.class : superClass;
         if (forceSuperClass || (superClass.equals(Object.class) && additionalInterfaces.isEmpty())) {
-            // No interface beans must use the bean impl as superclass
+            // No interface beans, must use the bean impl as superclass
             superClass = proxiedBeanType;
         }
         this.beanType = superClass;
 
         addDefaultAdditionalInterfaces();
         baseProxyName = proxyName;
+        // TODO can we expect that all services now support class defining? Throw an exception if they don't?
         proxyServices = Container.instance(contextId).services().get(ProxyServices.class);
```

Comment:
Depending on what we do in https://github.com/weld/api/pull/126, we might want to check this and throw in case someone provides custom `ProxyServices` that don't implement it. WDYT @mkouba ?